File: Models/Product.cs
namespace SolutanPerfumes.Models;

public class Product
{
    public string Id { get; set; } = "";
    public string Name { get; set; } = "";
    public string Description { get; set; } = "";
    public decimal Price { get; set; }
    /// <summary>Path under wwwroot, e.g. /images/products/perfume-1.jpg</summary>
    public string ImageUrl { get; set; } = "";
    public Category Category { get; set; }
    public bool Featured { get; set; }
}

File: src/types/product.ts
export type Category = 'perfumes' | 'makeup' | 'skincare';

export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  imageUrl: string;
  category: Category;
  featured?: boolean;
}

export interface CartItem {
  product: Product;
  quantity: number;
}

export interface Order {
  id: string;
  items: CartItem[];
  total: number;
  customerName: string;
  phone: string;
  address: string;
  paymentMethod: 'cod';
  status: 'pending' | 'confirmed' | 'shipped' | 'delivered';
  createdAt: string;
}
